service/Server.py

```diff
@@ -50,20 +56,48 @@ def __create_socket(self):
 			sys.exit(socket.error)
 
 	# def __create_files_dictionary(self):
-	#	for dir_entry in os.scandir('shared'):
-	#		self.files_dict[dir_entry.name] = {'md5': hasher.get_md5(dir_entry.path), 'size': dir_entry.stat().st_size}
+	# for dir_entry in os.scandir('shared'):
+	# self.files_dict[dir_entry.name] = {'md5': hasher.get_md5(dir_entry.path), 'size': dir_entry.stat().st_size}
+
+	def close_socket(self) -> None:
+		""" Close the passive socket ending the accept
+
+		:return: None
+		"""
+		try:
+			self.ss.shutdown(2)
+		except OSError:
+			self.ss.close()
```

Comment:
Sei sicuro che la shutdown chiuda anche il file descriptor? Se chiude solo gli estremi di lettura e scrittura, la close è corretta farla in ogni caso, non solo quando parte l'eccezione del mac.

Reply:
Comunque secondo me su linux esegue la shutdown e basta, perché genera subito l'errore ed esce, ho comunque aggiunto la close perché non si sa mai.